source/Octopus.Tentacle.Tests.Integration/Support/NugetTentacleFetcher.cs

```diff
@@ -0,0 +1,93 @@
+using System.Threading.Tasks;
+using NuGet.Frameworks;
+using NUnit.Framework;
+using Octopus.Tentacle.Util;
+
+namespace Octopus.Tentacle.Tests.Integration.Support
+{
+    public class NugetTentacleFetcher : ITentacleFetcher
+    {
+        static string DownloadUrlForVersion(string versionString) => $"https://f.feedz.io/octopus-deploy/dependencies/packages/Octopus.Tentacle.CrossPlatformBundle/{versionString}/download";
+
+        public async Task<string> GetTentacleVersion(string downloadPath, string version)
+        {
+            return await DownloadAndExtractFromUrl(downloadPath, DownloadUrlForVersion(version));
+        }
+
+        async Task<string> DownloadAndExtractFromUrl(string directoryPath, string url)
+        {
+            await Task.CompletedTask;
+            var downloadFilePath = Path.Combine(directoryPath, Guid.NewGuid().ToString("N"));
+
+            TestContext.WriteLine($"Downloading {url} to {downloadFilePath}");
+            await OctopusPackageDownloader.DownloadPackage(url, downloadFilePath);
+
+            // For local TODO delte this.
+            //File.Copy("/tmp/tentaclenuget/Octopus.Tentacle.CrossPlatformBundle.6.3.451.nupkg", downloadFilePath);
+
+            var extractionDirectory = Path.Combine(directoryPath, "extracted");
+
+            ZipFile.ExtractToDirectory(downloadFilePath, extractionDirectory);
+
+            var tentacleArtifact = Path.Combine(extractionDirectory, TentacleArtifactName());
+
+            var tentacleFolder = Path.Combine(directoryPath, "tentacle");
+            if (tentacleArtifact.EndsWith(".tar.gz"))
+            {
+                LinuxTentacleFetcher.ExtractTarGzip(tentacleArtifact, tentacleFolder);
+            }
+            else
+            {
+                ZipFile.ExtractToDirectory(tentacleArtifact, tentacleFolder);
+            }
+
+            return Path.Combine(tentacleFolder, "tentacle", "Tentacle");
+        }
+
+        public string TentacleArtifactName()
+        {
+            if (PlatformDetection.IsRunningOnWindows)
+            {
+                if (TentacleBinaryFrameworkForCurrentOs() == "net48")
+                {
+                    return "tentacle-net48-win.zip";
+                }
+                return $"tentacle-{TentacleBinaryFrameworkForCurrentOs()}-win-{Architecture()}.zip";
+            }
+
+            if (PlatformDetection.IsRunningOnMac)
+            {
+                return $"tentacle-{TentacleBinaryFrameworkForCurrentOs()}-osx-{Architecture()}.tar.gz";
+            }
+
+            if (PlatformDetection.IsRunningOnNix)
+            {
+                return $"tentacle-{TentacleBinaryFrameworkForCurrentOs()}-linux-{Architecture()}.tar.gz";
+            }
+
+            throw new Exception("Wow this is one fancy OS that tentacle probably can't run on");
+        }
+
+        public static string TentacleBinaryFrameworkForCurrentOs()
+        {
+            // This wont work for future versions of dotnet
```

Comment:
Yeah, it'd be good if we could invert this logic i.e. if its Framework then download `net48`, otherwise whatever the 'last dotnet version is

Reply:
In this case we have to download the entire 460MB nuget package which contains the tentacle bins for all OSs and architectures supported.

Only the linux and mac Tentacle binaries appear to be distributed on their own. I implemented a different ITentacleFetcher  for linux to take advantage of that which means on linux only ~35MB would ever need to be downloaded.